lib/dialects/abstract/query-generator.js

```diff
@@ -1320,12 +1320,18 @@ var QueryGenerator = {
                   }, topInclude.model);
                 }
 
-                options.where['__' + throughAs] = self.sequelize.asIs([
+                var queryWhere = self.sequelize.asIs([
                   '(',
                     $query.replace(/\;$/, ''),
                   ')',
                   'IS NOT NULL'
                 ].join(' '));
+
+                if (_.isPlainObject(options.where)) {
```

Comment:
As seen in lodash's doc, isPlainObject works for {}, isObject works for [], but I don't want arrays to be treated like an object. 
So isPlainObject, seems to be the good solution.